### PR TITLE
*: disable prepare cache when encounter cache table

### LINKED ISSUE #29641: disable prepare plan cache for cached table

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve?

Issue Number: close #29641 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by sylzd** [comment]
what about prepare a join b, when a is a cache table but b is a normal table...

File: expression/integration_test.go [hunk 1]
```diff
@@ -6207,6 +6207,29 @@ func (s *testIntegrationSerialSuite) TestPreparePlanCache(c *C) {
...
+	tk.Se, err = session.CreateSession4TestWithOpt(s.store, &session.Opt{
+		PreparedPlanCache: kvcache.NewSimpleLRUCache(100, 0.1, math.MaxUint64),
+	})
+	c.Assert(err, IsNil)
+	// Use the example from the docs https://docs.pingcap.com/tidb/stable/sql-prepare-plan-cache
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t(a int);")
+	tk.MustExec("alter table t cache")
+	tk.MustExec("prepare stmt from 'select * from t where a = ?';")
```
Comment (tiancaiamao) [inline expression/integration_test.go:6225]: You need to query for the first time and make sure the cache is loaded and used...

Comment (sylzd) [reply expression/integration_test.go:6225]: Okay~  I thought it was tested in `dml read operation`

File: planner/core/common_plans.go [hunk 2]
```diff
@@ -384,6 +384,19 @@ func (e *Execute) getPhysicalPlan(ctx context.Context, sctx sessionctx.Context,
 	sessVars := sctx.GetSessionVars()
 	stmtCtx := sessVars.StmtCtx
 	prepared := preparedStmt.PreparedAst
+	// disable the cache if cache table in prepared statement
+	for _, vInfo := range preparedStmt.VisitInfos {
```
Comment (tiancaiamao) [inline planner/core/common_plans.go:388]: Can we move it to line 414?   After the first time check, `prepared.UseCache` remain is false and not checked anymore?

If you put it here, the performance of every query is affected even you don't use the small cached table. @sylzd

**Review by tiancaiamao — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**